crates/deltalake-core/src/delta_datafusion/cdf/scan.rs

```diff
@@ -0,0 +1,318 @@
+use std::sync::Arc;
+
+use arrow_schema::{ArrowError, DataType, Field, SchemaRef, TimeUnit};
+use datafusion::datasource::file_format::parquet::ParquetFormat;
+use datafusion::datasource::file_format::FileFormat;
+use datafusion::datasource::physical_plan::FileScanConfig;
+use datafusion::execution::{SendableRecordBatchStream, TaskContext};
+use datafusion::physical_plan::coalesce_partitions::CoalescePartitionsExec;
+use datafusion::physical_plan::union::UnionExec;
+use datafusion::physical_plan::ExecutionPlan;
+use datafusion::prelude::SessionContext;
+use datafusion_common::{ScalarValue, Statistics};
+use lazy_static::lazy_static;
+
+use crate::delta_datafusion::cdf::scan_utils::*;
+use crate::delta_datafusion::register_store;
+use crate::kernel::{Action, Add, AddCDCFile, Remove};
+use crate::logstore::{get_actions, LogStoreRef};
+use crate::{DeltaResult, DeltaTableError};
+
+use super::*;
+
+pub const CHANGE_TYPE_COL: &str = "_change_type";
+pub const COMMIT_VERSION_COL: &str = "_commit_version";
+pub const COMMIT_TIMESTAMP_COL: &str = "_commit_timestamp";
+
+lazy_static! {
+    static ref CDC_PARTITION_SCHEMA: Vec<Field> = vec![
+        Field::new(COMMIT_VERSION_COL, DataType::Int64, true),
+        Field::new(
+            COMMIT_TIMESTAMP_COL,
+            DataType::Timestamp(TimeUnit::Millisecond, None),
+            true
+        )
+    ];
+    static ref ADD_PARTITION_SCHEMA: Vec<Field> = vec![
+        Field::new(CHANGE_TYPE_COL, DataType::Utf8, true),
+        Field::new(COMMIT_VERSION_COL, DataType::Int64, true),
+        Field::new(
+            COMMIT_TIMESTAMP_COL,
+            DataType::Timestamp(TimeUnit::Millisecond, None),
+            true
+        ),
+    ];
+}
+
+pub struct DeltaCdfScan {
+    log_store: LogStoreRef,
+    starting_version: i64,
+    schema: SchemaRef,
+    partition_values: Vec<String>,
+}
+
+impl DeltaCdfScan {
```

Comment:
Did not have time for a proper review, just as a quick comment - we should implement `ExecutionPlan` for this scan, so it can be used in datafusion as an operation.